cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -67,8 +79,8 @@
   private final long logicalStartTime;
   private final ProgramWorkflowRunnerFactory workflowProgramRunnerFactory;
   private NettyHttpService httpService;
-  private volatile boolean running;
-  private volatile WorkflowStatus workflowStatus;
+  private ExecutorService executor;
+  private ConcurrentHashMap<String, WorkflowStatus> status = new ConcurrentHashMap<String, WorkflowStatus>();
```

Comment:
Type can be just `Map` in this case as you are not using any method on the `ConcurrentMap` (e.g. you can replace the instance with a synchronized map and it will still works).